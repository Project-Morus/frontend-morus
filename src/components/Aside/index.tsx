import { ContainerSC, ContentSC, FooterSC } from "./styles";
import { IAsideProps } from "./types";
import LogoWhiteSvg from "../../assets/svg/LogoWhite";

const Aside = ({ position = 'right', ...props }: IAsideProps) => {
  const roleSVG = 'logoMorusWhite'

  return (
    <ContainerSC $position={position} role={props.role}>
      <ContentSC $position={position}>
        <LogoWhiteSvg role={roleSVG} />
      </ContentSC>

      {props.children}

      <FooterSC $position={position}>Todos os direitos reservados Morus Company 2023 | Vers√£o 1.0.0</FooterSC>
    </ContainerSC>
  );
}

export default Aside;