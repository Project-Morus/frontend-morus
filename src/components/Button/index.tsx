import { ReactNode } from "react";
import { IButtonProps, VariantsTypes } from "./types";
import ButtonPrimary from "./variants/Primary";
import ButtonSecondary from "./variants/Secondary";
import ButtonRed from "./variants/Red";
import ButtonGreen from "./variants/Green";

const Button = ({ variant = "primary", ...props }: IButtonProps) => {
  const variantType: Record<VariantsTypes, ReactNode> = {
    primary: <ButtonPrimary {...props} />,
    secondary: <ButtonSecondary {...props} />,
    red: <ButtonRed {...props} />,
    green: <ButtonGreen {...props} />,
  };

  return variantType[variant];
};

export default Button;
