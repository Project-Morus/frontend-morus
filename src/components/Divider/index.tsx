import { ReactNode } from "react";
import { IDividerProps, VariantsTypes } from "./types";
import DividerPrimary from "./variants/Primary";
import DividerSecondary from "./variants/Secondary";

const Divider = ({ variant = "primary", ...props }: IDividerProps) => {
  const variantType: Record<VariantsTypes, ReactNode> = {
    primary: <DividerPrimary {...props} />,
    secondary: <DividerSecondary {...props} />,
  };

  return variantType[variant];
};

export default Divider;
