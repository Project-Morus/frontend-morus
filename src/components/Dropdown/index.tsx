import { useState } from "react";
import { DropdownContainer, DropdownButton, DropdownMenu, DropdownItem, LabelContainer } from "./styles";
import { DropdownProps } from "./types";
import Icon from "../DataDisplay/Icon";
import { useTheme } from "styled-components";

function Dropdown({ label, options }: DropdownProps) {
  const [isOpen, setIsOpen] = useState(false);
  const theme = useTheme();

  const toggleDropdown = () => {
    setIsOpen(!isOpen);
  };

  return (
    <DropdownContainer>
      <DropdownButton onClick={toggleDropdown}>
        <LabelContainer>
          {label}
          {isOpen ? (
            <Icon name="ph-caret-up" size="16px" stroke="bold" color={theme.colors.primary[500]} />
          ) : (
            <Icon name="ph-caret-down" size="16px" color={theme.colors.primary[500]} />
          )}
        </LabelContainer>
      </DropdownButton>
      {isOpen && (
        <DropdownMenu>
          {options.map((item: string, index: number) => (
            <DropdownItem onClick={toggleDropdown} key={index}>
              {item}
            </DropdownItem>
          ))}
        </DropdownMenu>
      )}
    </DropdownContainer>
  );
}

export default Dropdown;
