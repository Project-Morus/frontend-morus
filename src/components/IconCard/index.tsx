import Icon from "../Icon";
import { CardSC } from "./styles";
import { IIconCardProps } from "./types";

const IconCard = ({ icon, stroke, variant = 'white', ...props }: IIconCardProps) => {

  return (
    <CardSC $variant={variant}
      role={props.role}
      className={props.className}
      data-testid={props?.['data-testid']}
    >
      <Icon
        name={icon}
        size="47"
        stroke={stroke}
      />
    </CardSC>
  );
}

export default IconCard;