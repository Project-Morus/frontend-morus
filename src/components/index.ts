
// Buttons
export { default as Button } from './Buttons/Button'
export { default as RadioButtonGroup } from './Buttons/RadioButtonGroup'
export { default as RadioButton } from './Buttons/RadioButton'

// DataDisplay
export { default as Icon } from './DataDisplay/Icon'
export { default as IconCard } from './DataDisplay/IconCard'
export { default as SequencialStepper } from './DataDisplay/SequencialStepper'
export { default as Table } from './DataDisplay/Table'

export { default as Archive } from './DataDisplay/Archive'

// Feedback
export { default as Aside } from './Feedback/Aside'
export { default as Divider } from './Feedback/Divider'
export { default as Modal } from './Feedback/Modal'


// Inputs
export { default as Input } from './Inputs/Input'

// Navigations
export { default as Menu } from './Navigation/Menu';
