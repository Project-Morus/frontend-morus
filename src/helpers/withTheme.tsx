import { ComponentType, JSX } from 'react';
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../styles/themes/default";

type ComponentPropsType<T> = T & JSX.IntrinsicAttributes

export function withTheme<T>(WrapperComponent: ComponentType<ComponentPropsType<T>>) {
  return (props: ComponentPropsType<T>) => {
    return (
      <ThemeProvider theme={defaultTheme}>
        <WrapperComponent {...props} />
      </ThemeProvider>
    );
  };
}

