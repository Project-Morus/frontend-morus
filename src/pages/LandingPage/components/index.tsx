export * from './InformationSection';
export * from './HeroSection';
export * from './CommonAreaSection';
export * from './CustomersSection';
export * from './FooterSection';