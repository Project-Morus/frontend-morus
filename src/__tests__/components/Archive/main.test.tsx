import { render, screen } from "@testing-library/react";
import { Archive } from "../../../components";
import { withTheme } from "../../../helpers/withTheme";

const ArchiveWithTheme = withTheme(Archive);

const role = "archive";

describe("Archive Component", () => {
  test("Verify if component is rendering", () => {
    render(<ArchiveWithTheme role={role} />);

    expect(screen.getByRole(role)).toBeVisible();
  })

  test("Snapshot Archive with variant default", () => {
    const { asFragment } = render(<ArchiveWithTheme role={role} />);
    expect(asFragment()).toMatchSnapshot();
  });
  test("Snapshot Archive with variant small", () => {
    const { asFragment } = render(<ArchiveWithTheme variant="small" role={role} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
