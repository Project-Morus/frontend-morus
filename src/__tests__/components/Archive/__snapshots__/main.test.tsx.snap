// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Archive Component Snapshot Archive with variant default 1`] = `
<DocumentFragment>
  .c1 {
  color: #3575B6;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: center;
  max-width: 100%;
  width: fit-content;
  width: -moz-fit-content;
  height: fit-content;
  height: -moz-fit-content;
}

.c0 {
  width: 800px;
  max-width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 1.5rem;
  border: 2px dashed #568CC2;
  padding: 3.5rem;
}

.c4 {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
}

.c3 {
  font-size: 1.8125rem;
  font-weight: 700;
  color: #999999;
}

.c5 {
  font-weight: 700;
  color: #999999;
}

.c7 {
  color: #F58317;
}

.c6 {
  color: #023B74;
}

<main
    class="c0"
    role="archive"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="ph-file-arrow-down"
      >
        <svg
          fill="currentColor"
          height="70px"
          viewBox="0 0 256 256"
          width="70px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,82.34l-56-56A8,8,0,0,0,152,24H56A16,16,0,0,0,40,40V216a16,16,0,0,0,16,16H200a16,16,0,0,0,16-16V88A8,8,0,0,0,213.66,82.34ZM160,51.31,188.69,80H160ZM200,216H56V40h88V88a8,8,0,0,0,8,8h48V216Zm-42.34-61.66a8,8,0,0,1,0,11.32l-24,24a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L120,164.69V120a8,8,0,0,1,16,0v44.69l10.34-10.35A8,8,0,0,1,157.66,154.34Z"
          />
        </svg>
      </div>
    </div>
    <h1
      class="c3"
    >
      Arraste e Solte ou Faça Upload dos Arquivos
    </h1>
    <div
      class="c4"
    >
      <span
        class="c5"
      >
        Suporta somente os formatos 
        <strong
          class="c6"
        >
          PDF
        </strong>
         e 
        <strong
          class="c6"
        >
          PNG
        </strong>
      </span>
      <span
        class="c5"
      >
        O tamanho limite dos arquivos é de 
        <strong
          class="c7"
        >
          50MB
        </strong>
      </span>
    </div>
  </main>
</DocumentFragment>
`;

exports[`Archive Component Snapshot Archive with variant small 1`] = `
<DocumentFragment>
  .c1 {
  color: #3575B6;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: center;
  max-width: 100%;
  width: fit-content;
  width: -moz-fit-content;
  height: fit-content;
  height: -moz-fit-content;
}

.c0 {
  width: 600px;
  max-width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  gap: 1.5rem;
  border: 2px dashed #568CC2;
  padding: 2.25rem;
}

.c4 {
  display: flex;
  flex-direction: column;
  gap: 0.5rem;
}

.c3 {
  font-size: 1.5rem;
  text-align: center;
  font-weight: 700;
  color: #999999;
}

.c5 {
  font-weight: 700;
  color: #999999;
}

.c7 {
  color: #F58317;
}

.c6 {
  color: #023B74;
}

<main
    class="c0"
    role="archive"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
        data-testid="ph-file-arrow-down"
      >
        <svg
          fill="currentColor"
          height="60px"
          viewBox="0 0 256 256"
          width="60px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M213.66,82.34l-56-56A8,8,0,0,0,152,24H56A16,16,0,0,0,40,40V216a16,16,0,0,0,16,16H200a16,16,0,0,0,16-16V88A8,8,0,0,0,213.66,82.34ZM160,51.31,188.69,80H160ZM200,216H56V40h88V88a8,8,0,0,0,8,8h48V216Zm-42.34-61.66a8,8,0,0,1,0,11.32l-24,24a8,8,0,0,1-11.32,0l-24-24a8,8,0,0,1,11.32-11.32L120,164.69V120a8,8,0,0,1,16,0v44.69l10.34-10.35A8,8,0,0,1,157.66,154.34Z"
          />
        </svg>
      </div>
    </div>
    <h3
      class="c3"
    >
      Arraste e Solte ou Faça Upload dos Arquivos
    </h3>
    <div
      class="c4"
    >
      <span
        class="c5"
      >
         Suporta somente os formatos 
        <strong
          class="c6"
        >
          PDF
        </strong>
         e 
        <strong
          class="c6"
        >
          PNG
        </strong>
      </span>
      <span
        class="c5"
      >
        O tamanho limite dos arquivos é de 
        <strong
          class="c7"
        >
          50MB
        </strong>
      </span>
    </div>
  </main>
</DocumentFragment>
`;
